### PR TITLE
🏗 Fix sourcemaps in single-pass

### PR DESCRIPTION
**Pull request by jridgewell**
This moves us to using MagicString to generate sourcemaps for the simple string transformations we're performing when concating single pass binaries together. With that sourcemap, we can remap it to the original location using the old sourcemap.

### CODE REVIEW

File: build-system/compile/single-pass.js [hunk 1]
```diff
@@ -559,57 +562,63 @@ exports.singlePassCompile = async function(entryModule, options) {
     .then(wrapMainBinaries)
     .then(intermediateBundleConcat)
     .then(eliminateIntermediateBundles)
-    .then(thirdPartyConcat)
-    .then(removeInvalidSourcemaps)
-    .catch(err => {
-      err.showStack = false; // Useless node_modules stack
-      return Promise.reject(err);
-    });
```
Comment (rsimha) [inline build-system/compile/single-pass.js]: Why were these lines removed?

File: build-system/compile/single-pass.js [hunk 2]
```diff
@@ -559,57 +562,63 @@ exports.singlePassCompile = async function(entryModule, options) {
...
+    .then(thirdPartyConcat);
```
Comment (rsimha) [inline build-system/compile/single-pass.js]: Can the steps after `compile` be run directly on the vinyl-fs files without having to write / rewrite? I was thinking you could consolidate `compile()` and `singlePassCompile()`, and inject these steps before the `gulp.dest('.')` at the end.

File: build-system/compile/single-pass.js (hunk 1 above)
Comment (jridgewell) [reply build-system/compile/single-pass.js]: The error stacks are useful, they pointed me to the exact line that was failing in this file.

File: build-system/compile/single-pass.js (hunk 2 above)
Comment (jridgewell) [reply build-system/compile/single-pass.js]: Yes, but I think that's best as a follow up PR.

File: build-system/compile/single-pass.js (hunk 1 above)
Comment (rsimha) [reply build-system/compile/single-pass.js]: It's true they might be useful when you're editing `single-pass.js`, but most of the time, they distract from actual code errors seen while building the runtime. Perhaps you can reinstate the lines here and remove them whenever you're editing this file?

File: build-system/compile/single-pass.js (hunk 2 above)
Comment (rsimha) [reply build-system/compile/single-pass.js]: SG. Maybe add a TODO?

File: build-system/compile/single-pass.js (hunk 1 above)
Comment (jridgewell) [reply build-system/compile/single-pass.js]: Done.

File: build-system/compile/single-pass.js [hunk 3]
```diff
@@ -688,11 +711,30 @@ function eliminateIntermediateBundles() {
       targets.push(createFullPath(extension.version));
     }
     targets.forEach(path => {
-      const {code} = babel.transformFileSync(path, {
+      const map = loadSourceMap(path);
+      function returnMapFirst(map) {
```
Comment (erwinmombay) [inline build-system/compile/single-pass.js]: could you explain why we need this?

Comment (jridgewell) [reply build-system/compile/single-pass.js]: Because I didn't save the map to a filesystem to load. The second param to `resorcery` is a loader callback that either has to return a sourcemap that will be remapped, or return nothing to treat the file as an original sourcemap.
